vm/parser.ts

```diff
@@ -933,16 +1153,22 @@ module J2ME {
     }
     
     getAllInterfaces(): ClassInfo [] {
+      if (this.allInterfaces) {
+        return this.allInterfaces;
+      }
```

Comment:
This is called ~ 630 times during the startup of some midlets. I agree that we should think of better ways to optimize interface collection, maybe using a custom built hash map.